src/Controller/GameController.php

```diff
@@ -17,12 +22,107 @@ public function __construct(GameManipulator $gameManipulator)
         $this->gameManipulator = $gameManipulator;
     }
 
-    public function random(): Response
-    {        
-        $game = $this->gameManipulator->createGame();
+    /**
+     * @ParamConverter("game", options={"mapping": {"hash": "hash"}})
+     */
+    public function index(Request $request, Game $game): Response
```

Comment:
There's too much logic in this controller action. A controller action will not have more than 40 lines in general.

Reply:
Will do later